xsdvalidate/xsd_validate_activity.go

```diff
@@ -0,0 +1,226 @@
+package xsdvalidate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.artefactual.dev/tools/temporal"
+)
+
+const XSDValidateActivityName = "xsd-validate-activity"
+
+type XSDValidateActivity struct{}
+
+func NewXSDValidateActivity() *XSDValidateActivity {
+	return &XSDValidateActivity{}
+}
+
+type XSDValidateActivityParams struct {
+	// DirectoryPath is the path of the files to be validated.
+	DirectoryPath string
+
+	// Pattern is the regular expression for files to validate.
+	Pattern string
```

Comment:
Why did you decide to use a directory and a pattern? The current use cases for this activity are not going to need that. It also worries me for the result payload size if multiple files with multiple errors are validated. I'd just pass the XML file and, if needed, call the activity multiple times to validate more than one XML. It will be responsibility of the caller to figure out how to find the XML paths, using regular expressions if needed, etc.

Reply:
Yeah, but payload limit issues will break the workflow, its about finding the compromise. I think for this case it's better to do one by one and don't worry about pattern issues.

Reply:
That makes sense... sounds good!

Reply:
(Fixed.)